### DISCUSSION TITLE
How to spec a java object in a function schema?

### DISCUSSION THREAD

**Discussion by Tillaert** [metosin/malli#854] [Q&A] [answered]
Is there a way to write a funciton schema that checks for a java type?

eg. 
```
(defn f [a b c]
{:malli/schem [:=> [:cat ??? :int :string] :int]})
```
Where `???` checks whether an object of the type `org.hyperledger.fabric.shim.ChaincodeStub` ?

**Comment by ikitommi** [comment, marked as answer]
For just validation, you can use `:fn`, which takes any predicate function, e.g. `[:fn (partial instance? org.hyperledger.fabric.shim.ChaincodeStub)]`.
